src/Core/src/Signer/SignerV4.php

```diff
@@ -270,6 +271,13 @@ private function convertBodyToStream(Request $request, \DateTimeImmutable $now,
             $contentLength = $body->length();
         }
 
+        if (!$this instanceof SignerV4ForS3) {
+            $request->setBody($body = StringStream::create($body));
+
+            return;
+        }
+        // Code above can be removed in version 2.0
```

Comment:
Is code below is only for S3?

Could we add a comment to say that it was moved to the S3 client?

Reply:
good idea. Done